### PR TITLE
Abstract class for collections

### PR DESCRIPTION
**Pull request by zzuga**
This is a new class to base our collection endpoints on.
It will avoid duplicate code between different descendants.

This will end up being the ancestor of wp-endpoints-collection and wp-endpoints-authors

### CODE REVIEW

**Comment by mitogh** [comment]
mmm I think the class should allow to overwrite the loop function otherwise is not going to be reusable.

**Comment by mitogh** [comment]
Also I think this is more part of the collection rather than the endpoint.

**Comment by zzuga** [comment]
@mitogh I have address your first comment, let me know if its not ok.

@adamf321 Cris is suggesting that I move this class to another package, what do you think?

File: src/AbstractCollectionEndpoint.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	 */
+	public function endpoint_callback( \WP_REST_Request $request ) {
+		$this->args = $request->get_params();
+		return $this->filter_data( $this->loop() );
+	}
+
+	/**
+	 * WP_Query Loop that has been triggered from the endpoint.
+	 */
+	protected function loop() {
```
Comment (mitogh) [inline src/AbstractCollectionEndpoint.php]: You don't need to specify the body of the function you just need to specify that is an abstract function.

**Comment by zzuga** [comment]
Im closing this one and creating a new project, we will follow up there

**Comment by mitogh** [comment]
@zzuga did you take a look at the discusson on the dev room?

**Comment by zzuga** [comment]
oops, reading. thanks

**Pull request closed without merging**